cpp/mg_utility/mg_graph.cpp

```diff
@@ -0,0 +1,152 @@
+#include "mg_graph.hpp"
+#include "mg_exceptions.hpp"
+#include <algorithm>
```

Comment:
Try separating includes into groups.
E.g. project headers and STL headers.
Just add new line before `#include <algorithm>`